cromwell_tools/tests/test_cromwell_tools.py

```diff
@@ -45,68 +69,54 @@ def setUp(self):
-        self.url = "https://fake_url"
-        self.user = "fake_user"
-        self.password = "fake_password"
-        self.caas_key = "path/fake_key.json"
+
+    def _start_workflows(self, cromwell_auth, mock_request, _request_callback):
+        mock_request.post(cromwell_auth.url + '/api/workflows/v1', json=_request_callback)
+        return cromwell_tools.start_workflow(
+            self.wdl_file, self.inputs_file, cromwell_auth, self.options_file,
+            self.inputs_file2, self.zip_file, label=self.label)
 
     @requests_mock.mock()
-    def test_start_workflow(self, mock_request):
+    @mock.patch('cromwell_tools.cromwell_tools.generate_auth_header_from_key_file')
+    def test_start_workflow(self, mock_request, mock_header):
         """Unit test using mocks
         """
+        mock_header.return_value = {"Authorization": "bearer fake_token"}
+
         def _request_callback(request, context):
             context.status_code = 200
             context.headers['test'] = 'header'
             return {'request': {'body': "content"}}
 
-        # Check request actions
-        mock_request.post(self.url, json=_request_callback)
-        result = cromwell_tools.start_workflow(
-            self.wdl_file, self.inputs_file, self.url, self.options_file, self.inputs_file2, self.zip_file, self.user,
-            self.password, label=self.label)
-        self.assertEqual(result.status_code, 200)
-        self.assertEqual(result.headers.get('test'), 'header')
+        for cromwell_auth in auth_options:
+            result = self._start_workflows(cromwell_auth, mock_request, _request_callback)
+            self.assertEqual(result.status_code, 200)
+            self.assertEqual(result.headers.get('test'), 'header')
 
     @requests_mock.mock()
     def test_start_workflow_retries_on_error(self, mock_request):
+
         def _request_callback(request, context):
             context.status_code = 500
             context.headers['test'] = 'header'
             return {'status': 'error', 'message': 'Internal Server Error'}
 
         # Make the test complete faster by limiting the number of retries
-        cromwell_tools.start_workflow.retry.stop = stop_after_attempt(3)
+        cromwell_tools.start_workflow.retry.stop = stop_after_attempt(2)
 
         # Check request actions
-        mock_request.post(self.url, json=_request_callback)
-        with self.assertRaises(requests.HTTPError):
-            result = cromwell_tools.start_workflow(
-                self.wdl_file, self.inputs_file, self.url, self.options_file, self.inputs_file2, self.zip_file,
-                self.user, self.password, label=self.label)
-            self.assertNotEqual(mock_request.call_count, 1)
+        for cromwell_auth in auth_options:
+            with self.assertRaises(requests.HTTPError):
+                _ = self._start_workflows(cromwell_auth, mock_request, _request_callback)
+                self.assertEqual(mock_request.call_count, 2)
```

Comment:
This line needs to be unindented (at the same level as the `with` statement) otherwise it will not raise an error in the case that this evaluates to false.